compiler/cbuilderstmts.nim

```diff
@@ -0,0 +1,23 @@
+template addAssignment(builder: var Builder, lhs: Snippet, valueBody: typed) =
+  builder.add(lhs)
+  builder.add(" = ")
+  valueBody
+  builder.add(";\n")
+
+template addFieldAssignment(builder: var Builder, lhs: Snippet, name: string, valueBody: typed) =
+  builder.add(lhs)
+  builder.add("." & name & " = ")
+  valueBody
+  builder.add(";\n")
+
+template addDerefFieldAssignment(builder: var Builder, lhs: Snippet, name: string, valueBody: typed) =
+  builder.add(lhs)
+  builder.add("->" & name & " = ")
+  valueBody
+  builder.add(";\n")
+
+template addSubscriptAssignment(builder: var Builder, lhs: Snippet, index: Snippet, valueBody: typed) =
+  builder.add(lhs)
+  builder.add("[" & index & "] = ")
+  valueBody
+  builder.add(";\n")
```

Comment:
I really like this way of doing CGen is so simple. 👍 

I also wonder if it is an opportunity to generate more clean and readable C,
like old Cgen was full of unneeded empty whitespaces and new lines everywhere.